### ISSUE TITLE
DCC Editor "Change Palette" cannot be closed by the window's X button

### ISSUE THREAD

**Issue by bbrucee** [OpenDiablo2/HellSpawner#299]
## Describe the bug
As title says, the DCC Editor "Change Palette" cannot be closed by the window's X button. When the X button is pressed the window closes but reopens

## To Reproduce
1. Open a DCC file
2. Click on DCC Editor menu dropdown
3. Select Change Palette
4. Attempt to X out of the window

## Expected behavior
Window closes with the behavior of "Don't use any palette" or "Exit"

## Screenshots

![Animation](https://user-images.githubusercontent.com/13605787/120736044-9273f980-c4b9-11eb-8c90-9a1399316790.gif)
(Screen flickers in gif are from recording method going wrong, unrelated to the bug)

**Comment by bbrucee** [comment]
Stumbled on to this bug while investigating the DCC widget. It looks like it is caused by selectPalette boolean not being set to false when window is closed by the X button

https://github.com/OpenDiablo2/HellSpawner/blob/master/hswindow/hseditor/hsdcceditor/dcc_editor.go#L83 
https://github.com/OpenDiablo2/HellSpawner/blob/c07ca9a8dc1f1a23ab2cae2dba771a9600238994/hswidget/select_palette_widget.go#L45 (closeCB does not fire)

**Comment by gravestench** [comment]
Good catch, and thanks for reporting. After looking into this, I don't think that this is going to be an easy bug to fix with the current implementation.
